bugsnag-android-core/src/main/java/com/bugsnag/android/Event.java

```diff
@@ -419,7 +419,8 @@ void setProjectPackages(Collection<String> projectPackages) {
                 boolean unhandled,
                 Metadata globalMetadata) {
             Throwable exc = unhandled ? exception : null;
-            this.threadState = new ThreadState(config, thread, Thread.getAllStackTraces(), exc);
+            // unhandled errors use the exception trace for thread traces
```

Comment:
Comment no longer required here as you've moved the if-else clause back to the `ThreadState`.